### ISSUE TITLE
Extend installation docs to include RHEL/CentOS instructions

### ISSUE THREAD

**Issue by jeremystretch** [digitalocean/netbox#447]
The installation docs currently assume a Debian-based Linux OS. We should annotate the equivalent commands for RHEL-based OSes as well.

**Comment by systo** [comment]
From my installation notes, please verify.  Install was done on a fresh install of centos 7 minimal.  Install assumes no root password to enable sudo, or sudo enabled and your user added to group wheel.

## For the PostgreSQL enable the epel repos
sudo yum install -y epel-release
sudo yum install -y postgresql postgresql-server postgresql-libs postgresql-devel

## For the Netbox installation portion
sudo yum install -y python27 python-devel git python-pip libxml2-devel libxslt-devel libffi-devel graphviz.x86_64

**Comment by marvnrawley** [comment]
I just installed it on CentOS 6, which was significantly more complicated than for 7, and would be happy to share notes if it will help.

**Comment by jeremystretch** [comment]
Per #441, it looks like `openssl-devel` should be included as well, to ensure that the Cryptography Python package compiles.

**Comment by marvnrawley** [comment]
Coming back to this with a couple of notes.

**Postgres**

CentOS 6 ships with PGSql 8.4 which is EoS and in fairness I'm not sure which version ships with CentOS 7. EPEL is a pretty standard addon for most CentOS builds however I'd probably suggest using the pgdg repo from PG themselves:

http://yum.postgresql.org/repopackages.php

I've got 9.5.x up and running quite happily with this.

**CentOS 6 in general**

- Python version. CentOS 6.x requires the system version of Python to be 2.6.x for yum. Yum doesn't work against Python2.7 so you have to run sidealong. I did this using SCL; but you then get all kinds library and binary path issues. For me, I appended it to the end, and largely got around it by doing the build by specifying pip2.7 install and it worked.
- As discussed above, the missing openssl-devel, it's not in by default on CentOS 6.
- Due I think mostly to the path issues with python I struggled to get gunicorn/supervisord going. I believe they were trapped in the system default bin/library and I couldn't decipher/didn't want to really mess around with that in case I broke Yum.
- I ended up with mod_wsgi in Apache2.4, which works fine, but requires 'WSGIPythonPath' to be added into the global config i.e. not inside the vhost. I'm not sure how this would scale in anything other than a 'dedicated-to-netbox-only' hosting environment, and given I'm probably a fairly typical user, I stuff a few open source utilities on a box (or in my case, a postgresql master-slave cluster with mirrored apache config & applications in warm standby on the second box).
- You can run mod_wsgi under Apache2.2 but Apache2.4 is preferred by the project, and I'm using it because SCL provides PHP5.4+ linked against Apache2.4 but not Apache2.2 and I also needed PHP for other tools. In order to get Apache2.4 up and running, I essentially had to symlink all the bins from the SCL directories into the usual locations, and also /etc/httpd/ became a symlink.
- To get mod_wsgi up and running, I used pip2.7 to install it and it just linked correctly against Apache2.4 after that.

In summary: it's quite a lot of faff, bypassing, upgrading system packages and general not-easy-to-maintain-ness. I'd probably recommend using a newer version of CentOS as a base; CentOS 6.x will be supported for another 3 - 4 years but the versions of key apps like Python, PHP, etc are not supported officially by their projects and it's going to cause a cascading headache for someone wanting to maintain the system.

**Comment by chagara** [comment]
I can make a PR with the doc I made before. That PR I submitted before was before the docs got separated. I can just add the centos part in each doc if you want.

**Comment by Snozzberries** [comment]
Here is the commands I used to get NetBox and dependencies installed on CentOS 7. This isn't fully complete as the service responds with a 503, so I need to figure that out.

# Install Postgresql
yum install postgresql postgresql-devel python-psycopg2 postgresql-server

# Configure Postgresql
useradd postgres
postgresql-setup initdb
systemctl start postgresql
systemctl enable postgresql
vi /var/lib/pgsql/data/pg_hba.conf
## Replace ident with md5
systemctl restart postgresql
sudo -u postgres psql
CREATE DATABASE netbox;
CREATE USER netbox WITH ENCRYPTED PASSWORD 'L0ckD0wn';
GRANT ALL PRIVILEGES ON DATABASE netbox TO netbox;
\q
psql -U netbox -h localhost -W
\q

# Install dependencies (Python 2.7 is installed by default)
yum install gcc python-devel libxml2-devel libxslt-devel libffi-devel graphviz postgresql-libs openssl-devel
curl "https://bootstrap.pypa.io/get-pip.py" -o "get-pip.py"
python get-pip.py
pip -V
rm -f get-pip.py

# Install Netbox (Update version)
yum install wget
wget https://github.com/digitalocean/netbox/archive/v1.5.2.tar.gz
tar -xzf v1.5.2.tar.gz -C /opt
rm -f v1.5.2.tar.gz
cd /opt/
ln -s netbox-1.5.2/ netbox
cd /opt/netbox/

# Install Python Packages
pip install --index-url=http://pypi.python.org/simple/ --trusted-host pypi.python.org -r requirements.txt

# Configure NetBox
cd netbox/netbox/
cp configuration.example.py configuration.py
vi configuration.py
## Add ALLOWED_HOSTS
## Add user and password from above
## Generate a 50 character alpha and numeric for secrect (Wolfram Alpha)

# Migrate DB
cd ..
./manage.py migrate

# Create super user
./manage.py createsuperuser

# Collect Static
./manage.py collectstatic

# Load initial data
./manage.py loaddata initial_data

# Install remaining
pip install gunicorn
pip install supervisor
yum install httpd
mkdir /etc/httpd/sites-enabled
mkdir /etc/httpd/sites-available
vi /etc/httpd/sites-available/nextbox.conf
## Use XML from config guide
cd /etc/httpd/sites-enabled/
ln -s /etc/httpd/sites-available/netbox.conf
vi /etc/httpd/conf/httpd.conf
## Include sites-enabled/*.conf
## LoadModule proxy_module modules/mod_proxy.so
## LoadModule proxy_http_module modules/mod_proxy_http.so
systemctl restart httpd

cd /opt/netbox
vi gunicorn_config.py
## Use config from guide

echo_supervisord_conf > supervisord.conf
mv supervisord.conf /etc/supervisord.conf
mkdir /etc/supervisord.d/
vi /etc/supervisord.conf
## Edit files = /etc/supervisord.d/*.conf

vi /etc/rc.d/init.d/supervisord
## Use the init.d script from https://rayed.com/wordpress/?p=1496
chmod +x /etc/rc.d/init.d/supervisord
chkconfig --add supervisord
chkconfig supervisord on
service supervisord start
vi /etc/supervisord.d/netbox.conf
## Use the supervisor script from the guide
service supervisord restart

**Comment by jeremystretch** [comment]
I've adapted @chagara's PR to include CentOS/RHEL instructions for installing PostgreSQL, Django, and NetBox. I'm going to leave the web server portion focused on Debian/Ubuntu because we're already including both nginx and Apache, and I don't want the documentation to become to cumbersome.
